package com.projeto.sistema_chamado.comentario;

import com.projeto.sistema_chamado.chamado.domain.Chamado;
import com.projeto.sistema_chamado.usuario.Usuario;
import jakarta.persistence.*;

import java.time.Instant;

@Entity
@Table(name = "comentario")
public class Comentario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "chamado_id", nullable = false)
    private Chamado chamado;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "autor_id", nullable = false)
    private Usuario autor;

    @Column(nullable = false, columnDefinition = "text")
    private String texto;

    @Column(name = "criado_em", nullable = false)
    private Instant criadoEm;

    protected Comentario() {
    }

    public Comentario(
            Chamado chamado,
            Usuario autor,
            String texto,
            Instant criadoEm) {

        this.chamado = chamado;
        this.autor = autor;
        this.texto = texto;
        this.criadoEm = criadoEm;
    }

    public Long getId() {
        return id;
    }

    public Chamado getChamado() {
        return chamado;
    }

    public Usuario getAutor() {
        return autor;
    }

    public String getTexto() {
        return texto;
    }

    public Instant getCriadoEm() {
        return criadoEm;
    }
}
